src/main/java/seedu/address/logic/commands/removecommand/RemoveMentorCommand.java

```diff
@@ -50,12 +50,12 @@ public CommandResult execute(Model model) throws CommandException {
 
         try {
             model.removeMentorFromTeam(teamId, mentorToBeRemoved);
-            model.updateHistory(this);
-            model.recordCommandExecution(this.getCommandInputString());
         } catch (AlfredException e) {
             throw new CommandException(MESSAGE_TEAM_DOES_NOT_HAVE_MENTOR);
         }
 
+        model.updateHistory(this);
+        model.recordCommandExecution(this.getCommandInputString());
```

Comment:
Should provide some context on this move

Reply:
Is it to make it always run?

Reply:
it's moved outside of the try-catch block because model history and command history are only updated when the command is successfully executed.